datafusion/src/execution/context.rs

```diff
@@ -639,6 +641,7 @@ impl Default for ExecutionConfig {
     fn default() -> Self {
         Self {
             concurrency: num_cpus::get(),
+            partitions: num_cpus::get(),
```

Comment:
For backwards compatibility what would you think about defaulting `partitions` to `None` and if it was not set using the value for `concurrency` instead? Otherwise users may have to tune two separate knobs when they may just want a big hammer "how many cores should data fusion try and keep busy"